internal/cache/cache.go

```diff
@@ -0,0 +1,111 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+//go:generate mockgen -source=cache.go -package=cache -destination=mock_cache.go
+
+type Cache interface {
+	DeleteExpired()
+	Get(key string) (interface{}, bool)
+	Set(key string, value interface{})
+	StopCleanup()
+}
+
+type Item struct {
+	Object interface{}
+
+	Expiration int64
```

Comment:
How about making this a `time.Time`? It would simplify some manipulations that currently use `UnixNano()`.

Reply:
We could definitely do that. The reason I chose an `int64` is to keep memory consumption as low as possible, but I see your point. Does it make sense memory wise or should we just refactor it to `time.Time` as you suggested?

Reply:
It [appears to make little sense](https://stackoverflow.com/questions/56963226/how-many-bytes-are-in-a-golang-time-object). I would use `time.Time` and switch to `int64` if we ever have a problem with memory consumption.